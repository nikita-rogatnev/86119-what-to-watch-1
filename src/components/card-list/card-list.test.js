import React from "react";
import Enzyme, {shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";

import CardList from "./card-list";

import mockData from "../../mocks/mock-data";

Enzyme.configure({adapter: new Adapter()});

it(`CardList renders correctly`, () => {
  const tree = shallow(<CardList
    data={mockData}
    showPlayButton={false}
  />);

  expect(toJson(tree)).toMatchSnapshot();
});
