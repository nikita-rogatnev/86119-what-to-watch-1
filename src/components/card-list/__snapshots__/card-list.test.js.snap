// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList renders correctly 1`] = `
<div
  className="catalog__movies-list"
>
  <Card
    backgroundColor="#B9B27E"
    backgroundImage="https://es31-server.appspot.com/wtw/static/film/background/matrix.jpg"
    description="A computer hacker learns from mysterious rebels about the true nature of his reality and his role in the war against its controllers."
    director="Wachowski Brothers"
    genre="Action"
    id={1}
    isFavorite={false}
    key="1"
    name="Matrix"
    posterImage="https://es31-server.appspot.com/wtw/static/film/poster/matrix.jpg"
    previewImage="https://es31-server.appspot.com/wtw/static/film/preview/matrix.jpg"
    previewVideoLink="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
    rating={8.7}
    released={1999}
    runTime={136}
    scoresCount={1503092}
    showPlayButton={false}
    starring={
      Array [
        "Keanu Reeves",
        "Laurence Fishburne",
        "Carrie-Anne Moss",
      ]
    }
    videoLink="http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4"
  />
  <Card
    backgroundColor="#F0DBA2"
    backgroundImage="https://es31-server.appspot.com/wtw/static/film/background/Johnny_English.jpg"
    description="After a sudden attack on the MI5, Johnny English, Britain's most confident yet unintelligent spy, becomes Britain's only spy."
    director="Peter Howitt"
    genre="Comedy"
    id={2}
    isFavorite={false}
    key="2"
    name="Johnny English"
    posterImage="https://es31-server.appspot.com/wtw/static/film/poster/Johnny_English.jpg"
    previewImage="https://es31-server.appspot.com/wtw/static/film/preview/johnny-english.jpg"
    previewVideoLink="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
    rating={6.2}
    released={2003}
    runTime={88}
    scoresCount={136843}
    showPlayButton={false}
    starring={
      Array [
        "Rowan Atkinson",
        "John Malkovich",
        "Natalie Imbruglia",
      ]
    }
    videoLink="http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4"
  />
</div>
`;
